src/core/document.js

```diff
@@ -578,30 +578,50 @@ class Page {
     return tree;
   }
 
-  getAnnotationsData(intent) {
-    return this._parsedAnnotations.then(function (annotations) {
-      const annotationsData = [];
+  async getAnnotationsData(handler, task, intent) {
+    const annotations = await this._parsedAnnotations;
+    if (annotations.length === 0) {
+      return [];
+    }
+
+    const partialEvaluator = new PartialEvaluator({
+      xref: this.xref,
+      handler,
+      pageIndex: this.pageIndex,
+      idFactory: this._localIdFactory,
+      fontCache: this.fontCache,
+      builtInCMapCache: this.builtInCMapCache,
+      standardFontDataCache: this.standardFontDataCache,
+      globalImageCache: this.globalImageCache,
+      options: this.evaluatorOptions,
+    });
+    const textContentPromises = [];
+    const annotationsData = [];
 
-      if (annotations.length === 0) {
-        return annotationsData;
+    const intentAny = !!(intent & RenderingIntentFlag.ANY),
+      intentDisplay = !!(intent & RenderingIntentFlag.DISPLAY),
+      intentPrint = !!(intent & RenderingIntentFlag.PRINT);
+
+    for (const annotation of annotations) {
+      // Get the annotation even if it's hidden because
+      // JS can change its display.
+      if (
+        intentAny ||
+        (intentDisplay && annotation.viewable) ||
+        (intentPrint && annotation.printable)
+      ) {
+        annotationsData.push(annotation.data);
       }
-      const intentAny = !!(intent & RenderingIntentFlag.ANY),
-        intentDisplay = !!(intent & RenderingIntentFlag.DISPLAY),
-        intentPrint = !!(intent & RenderingIntentFlag.PRINT);
 
-      for (const annotation of annotations) {
-        // Get the annotation even if it's hidden because
-        // JS can change its display.
-        if (
-          intentAny ||
-          (intentDisplay && annotation.viewable) ||
-          (intentPrint && annotation.printable)
-        ) {
-          annotationsData.push(annotation.data);
-        }
+      if (intentAny || (intentDisplay && annotation.viewable)) {
+        textContentPromises.push(
+          annotation.extractTextContent(partialEvaluator, task, this.view)
```

Comment:
We need to ensure that a possible Error in the new method won't break all Annotation-parsing here, so something like the following is probably necessary.
```suggestion
          annotation
            .extractTextContent(partialEvaluator, task, this.view)
            .catch(function (reason) {
                warn(
                  `getAnnotationsData - ignoring textContent during "${task.name}" task: "${reason}".`
                );
             })
```